src/loginServices.js

```diff
@@ -248,8 +248,8 @@ function handleLoginError(dispatch, resp) {
   dispatch(setOkapiReady());
 }
 
-function processOkapiSession(okapiUrl, store, tenant, resp) {
-  const token = resp.headers.get('X-Okapi-Token');
+function processOkapiSession(okapiUrl, store, tenant, resp, origToken) {
+  const token = resp.headers.get('X-Okapi-Token') || origToken || store.getState().okapi.token;
```

Comment:
I don't think we need ` || store.getState().okapi.token` here. This function is called from two places, `requestLogin` and `requestUserWithPerms`. In the former, the `X-Okapi-Token` header will be present. In the latter your change explicitly provides the value provided to `origToken` and that function is only called from `src/components/SSOLanding` when the token value is truthy.